clang/test/CodeGenSYCL/integration_header.cpp

```diff
@@ -1,4 +1,4 @@
-// RUN: %clang -I %S/Inputs -fsycl-device-only -Xclang -fsycl-int-header=%t.h %s -c -o %T/kernel.spv
+// RUN: %clang -I %S/Inputs -fsycl-device-only -Xclang -fsycl-int-header=%t.h %s -fsyntax-only -o %T/kernel.spv
```

Comment:
```suggestion
// RUN: %clang -I %S/Inputs -fsycl-device-only -Xclang -fsycl-int-header=%t.h %s -fsyntax-only
```
I would also add following TODOs for the follow-up improvements:
- I would also bypass the driver and use %clang_cc1.
- It's looks strange the we use `-fsyntax-only` in CodeGen test. I think we should address this by moving integration header generation to CodeGen library.

Reply:
@Fznamznon for the TODOs.

Do we have a good way of moving it architected.  I cannot think of a good way to move integration headers generation later in the process.

As far as using fsyntax-only in the CodeGen test, yes, this is weird.  I'm not convinced that this test should BE a CodeGen test as opposed to a SEMA test.

Reply:
I can't say if there is a good way to move integration header generation to the CodeGen or not, since I don't know all parts of the clang good enough. But I wouldn't refuse considering it.
IMO **generation of some code** in Sema seems weird for me (it has been weird ever since it appeared, but I didn't come up other suggestion so here it is). See https://clang.llvm.org/docs/InternalsManual.html#the-sema-library, it says that Sema does semantic analysis and builds AST. 
I think we can create an issue to investigate this approach (maybe asking the community about it) at least, instead of leaving TODO in the test that no one ever looks at.

Reply:
You're right that conceptually it could be in CodeGen (though, most call Clang Codegen "IR-Gen", since it doesn't generate 'code', it generates IR).

But I just don't see a good way to do it.  And splitting it from the rest of the kernel generation(which it is so closely related to) is perhaps a bad idea as well.  We should ask community when we come to open sourcing it, though I'm unconvinced "CodeGen" is the right place.